database/factories/SignatureFactory.php

```diff
@@ -15,6 +15,7 @@ public function definition()
     {
         return [
             'room_id'           => randomOrCreate(app('room')),
+            'location_id'       => randomOrCreate(app('location')),
```

Comment:
Adding location directly to signature eliminates the dependency on room to determine location.